### PR TITLE
Sophisticated investor

### PR DESCRIPTION
**Pull request by ejjick**
Sophisticated investor check. File upload interface for sophisticated investor documents. Uploads the file to an encrypted S3 bucket and provides approve/decline functionality in admin as well as direct links to the documents.

Caveats:
1. Deployment should be good but might be worth running in staging first
2. Local dev requires bucket creation (added to `runserver.sh`)
3. Code coverage is at 94.88% even though less is reported (codecov is down atm)

### CODE REVIEW

File: web/user_accounts/admin.py [hunk 1]
```diff
@@ -54,3 +59,71 @@ class EthereumAccountAdmin(admin.ModelAdmin):
...
+            action_flag=CHANGE,
+            change_message=_('Declined')
+        )
+    self.message_user(
+        request,
+        "%s sophisticated investor document(s) declined" % affected_rows)
+    email_content = render_to_string(
+        'user_accounts/email/si_declined.txt'
+    )
+    send_mail(
```
Comment (dpnova) [inline web/user_accounts/admin.py]: Can we not use fail silently here? I'd rather report something to sentry than have it gobbled up.

File: web/user_accounts/models.py [hunk 2]
```diff
@@ -130,3 +134,25 @@ def completed(self):
             return {'completed': False, 'reason': 'eth_address is None'}
 
         return {'completed': True, 'reason': None}
+
+
+def rename_file(instance, filename):
```
Comment (dpnova) [inline web/user_accounts/models.py]: This will barf if the file has no extension.

File: web/user_accounts/admin.py (hunk 1 above)
Comment (cballantyne) [reply web/user_accounts/admin.py]: Fixed up to log an error ... which in turn gets reported to sentry.

File: web/user_accounts/models.py (hunk 2 above)
Comment (cballantyne) [reply web/user_accounts/models.py]: It actually doesn't. I was surprised too. If no extension the new filename has no extension either. I have put a couple of tests around it to make sure working as expected.

Comment (ejjick) [reply web/user_accounts/models.py]: No extension makes the form invalid too. Thanks for testing though.

**Pull request merged**